File: Sem4/CN/Code/Java/ByteStuffing.java
import java.util.Scanner;

public class ByteStuffing {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n; 
        System.out.println("Enter the number of frames:\t"); 
        n = sc.nextInt();
        
        char[][] data = new char[n][20];
 
        for (int i = 0; i < n; i++) { 
            System.out.print("\nEnter frame "+(i+1)+" data\t"); 
            data[i][0] = sc.next().charAt(0);
        } 
 
        System.out.print("\nData sent from the sender's side is:\t"); 
        
        for(int i=0 ; i<n ;i++){ 
            System.out.print("$"); 

            for (int j = 0; j < data[i].length; j++) { 
                if(data[i][j] == '$' || data[i][j] == '#'){ 
                    System.out.print("#"); 
                } 
                System.out.print(data[i][j]); 
            } 

            System.out.print("$"); 
        } 

        sc.close();
    }
}

File: Sem4/CN/Code/Prac/ByteStuffing.c
#include <stdio.h>
#include <string.h>

int main(){
    int i, j, n, len;
    char send[n][100], receive[100];

    printf("***********Sender***********\n");
    
    printf("Enter the no. of frames: ");
    scanf("%d", &n);

    for (i = 0; i<n; i++){
        printf("Frame %d: ",(i+1));
        scanf("%s",&send[i]);
    }

    printf("\nData to be sent: \n");
    for (int i = 0; i<n; i++){
        printf("$");
        for (j=0; j<strlen(send[i]); j++){
            if (send[i][j]=='$' || send[i][j]=='#'){
                printf("#");
            }
            printf("%c",send[i][j]);
        }
        printf("$");
    }

    printf("\n\n***********Receiver***********\n");

    printf("\nEnter the received data: ");
    scanf("%s",receive);

    len=strlen(receive);
    i=1;

    while(i<len-1){
        printf("\n");
        do{
            if(receive[i] == '#'){
                i++;
            }
            printf("%c",receive[i]);
            i++;
        }while(receive[i] != '$');
        i++;
        if(receive[i] == '$'){
            i++;
        }
        else{
            break;
        }
    }
}